src/userent.c

```diff
@@ -776,10 +776,17 @@ int xtra_set(struct userrec *u, struct user_entry *e, void *buf)
     nfree(old->key);
     nfree(old->data);
     nfree(old);
+    if (old == e->u.extra) {
+      e->u.extra = NULL;
+    }
   }
   if (old != new && new->data) {
-    if (new->data[0])
+    if (new->data[0]) {
```

Comment:
Am I seeing this correct: if new->data is an empty string, it is not inserted at the place of "old", but we did free "old", leaving a gap?